diff --git a/src/build_targets.rs b/src/build_targets.rs
index 6cb554b..52d936f 100644
--- a/src/build_targets.rs
+++ b/src/build_targets.rs
@@ -89,7 +89,7 @@ impl BuildTargets {
                 let shared_lib = targetdir.join(format!("lib{lib_name}.so"));
                 (shared_lib, static_lib, None, None)
             }
-            ("macos", _) | ("ios", _) => {
+            ("macos", _) | ("ios", _) | ("tvos", _) => {
                 let static_lib = targetdir.join(format!("lib{lib_name}.a"));
                 let shared_lib = targetdir.join(format!("lib{lib_name}.dylib"));
                 (shared_lib, static_lib, None, None)
diff --git a/src/install.rs b/src/install.rs
index c087e74..53a968d 100644
--- a/src/install.rs
+++ b/src/install.rs
@@ -91,7 +91,7 @@ impl LibType {
             | ("haiku", _)
             | ("illumos", _)
             | ("emscripten", _) => LibType::So,
-            ("macos", _) | ("ios", _) => LibType::Dylib,
+            ("macos", _) | ("ios", _) | ("tvos", _) => LibType::Dylib,
             ("windows", _) => LibType::Windows,
             _ => unimplemented!("The target {}-{} is not supported yet", os, env),
         }
